src/absil/illib.fs

```diff
@@ -523,6 +524,116 @@ module ResultOrException =
         | Result x -> success x
         | Exception _err -> f()
 
+[<RequireQualifiedAccess>] 
+type ValueOrCancelled<'TResult> =
+    | Value of 'TResult
+    | Cancelled of OperationCanceledException
```

Comment:
Wondering if the OperationCanceledException will always contain info we would look for in context of diagnostic/troubleshooting?

If it is not the case, we could store an additional value (a string?) which would contain a message that we can lookup in our code.